camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/common/EtlCounts.java

```diff
@@ -241,18 +241,30 @@ private void produceCount(String brokerList, ArrayList<byte[]> monitorSet) {
     Producer producer = new Producer(new ProducerConfig(props));
     try {
       for (byte[] message : monitorSet) {
-        KeyedMessage keyedMessage = new KeyedMessage("TrackingMonitoringEvent", message);
-        producer.send(keyedMessage);
+        for (int i = 0; i <= publishCountMaxRetries; i++) {
+          try {
+            KeyedMessage keyedMessage = new KeyedMessage("TrackingMonitoringEvent", message);
+            producer.send(keyedMessage);
+            break;
+          } catch (Exception e) {
+            log.error("Publishing count for topic " + topic + " to " + brokerList.toString() + " has failed " + (i + 1)
+                + " times. " + (publishCountMaxRetries - i) + " more attempts will be made.");
+            if (i == publishCountMaxRetries) {
+              throw new RuntimeException(e);
```

Comment:
You might want to provide a error message (e.g. "max tried three times on publishing ...") together with the cause exception.